units/DAA0206/DAA0206_script.lua

```diff
@@ -16,16 +16,16 @@ DAA0206 = Class(AAirUnit) {
     Weapons = {
         Suicide = Class(DefaultProjectileWeapon) {}
     },
-    
+
     OnRunOutOfFuel = function(self)
-        self:Kill()
+        self.Kill(self)
     end,
     
     ProjectileFired = function(self)
-        self:GetWeapon(1).IdleState.Main = function(self) end
-        self:PlayUnitSound('Killed')
-		self:PlayUnitSound('Destroyed')
-        self:Destroy()  			
+        self.GetWeapon(self, 1).IdleState.Main = function(self) end
+        self.PlayUnitSound(self, 'Killed')
```

Comment:
These are moho functions - make them an upvalue at the top of the file (see projectile.lua inside lua/sim for an example). That allows us to skip the table operation (self.function)